rmw_opensplice_cpp/src/functions.cpp

```diff
@@ -1083,6 +1090,40 @@ rmw_take(const rmw_subscription_t * subscription, void * ros_message, bool * tak
   return RMW_RET_OK;
 }
 
+rmw_ret_t
+rmw_take(const rmw_subscription_t * subscription, void * ros_message, bool * taken)
+{
+  return _take(subscription, ros_message, taken, nullptr);
+}
+
+rmw_gid_t *
+rmw_take_with_gid(const rmw_subscription_t * subscription, void * ros_message, bool * taken)
```

Comment:
How does this function signal the specific error code? Shouldn't it return `rmw_ret_t` and have an additional argument `rmw_gid_t * &`?